AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m

```diff
@@ -83,12 +86,23 @@ - (void)setUserId:(nullable NSString *)userId {
      * crashes on apps between previous userId and current userId.
      */
     [self.userIdHistory removeLastObject];
+    BOOL sameUserId = (userId && [self.currentUserIdInfo.userId isEqualToString:(NSString *)userId]) ||
+                      (!userId && self.currentUserIdInfo.userId == userId);
     self.currentUserIdInfo.userId = userId;
     self.currentUserIdInfo.timestamp = [NSDate date];
     [self.userIdHistory addObject:self.currentUserIdInfo];
     [MS_USER_DEFAULTS setObject:[NSKeyedArchiver archivedDataWithRootObject:self.userIdHistory] forKey:kMSUserIdHistoryKey];
     MSLogVerbose([MSAppCenter logTag], @"Stored new userId:%@ and timestamp: %@.", self.currentUserIdInfo.userId,
                  self.currentUserIdInfo.timestamp);
+    if (sameUserId) {
```

Comment:
Question: if it's the same userId, wouldn't we want to return immediately instead of adding it to `self.userIdHistory` and writing to `MS_USER_DEFAULTS` ([line 93-96](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1487/files#diff-5550fe8ea22f6b4d65cedb980135be42R93)), right? 

It also seems wrong to log "Stored new userId..." message when we know it's the same UserId.

Reply:
Fixed now